### ISSUE TITLE
Items on 1.21 not stacking with other items

### ISSUE THREAD

**Issue by Mcklp** [apavarino/Deadchest#80]
Items on 1.21 have different nbt data making them impossible to stack with other items
- Deadchest
- 4.21.0
- 1.21
- This server is running Paper version 1.21-15-master

**Describe the bug**
Items from deadchest have one additional component listed after hovering on them, making it impossible to stack with items that are not from deadchest. Using command "/data get entity @s SelectedItem" they have "minecrfat:damage": 0 (i added a photo). 
Despite that deadchest work for me on 1.21 

**To Reproduce**
1. Get 2 same stackable items
2. Get killed with one of it
3. Pickup dead chest
4. Try to stack them

**Expected behavior**
Items from dead chest to be the same as the rest of the items.

**Screenshots**
![image](https://github.com/apavarino/Deadchest/assets/47704974/2bd8799b-a76e-45d1-bddc-785641a36fdd)

**Comment by Mcklp** [comment]
Yeah, thats right @leshicodes, but i wanted to write my observations to maybe help with updateing it to 1.21. I know that this version isnt supported and thats why something like that happened

**Comment by Mcklp** [comment]
Yeah, thats right @leshicodes, but i wanted to write my observations to maybe help with updateing it to 1.21. I know that this version isnt supported and thats why something like that happened.

**Comment by leshicodes** [comment]
@Mcklp very fair, and I had a feeling that's what your intentions were. I am a novice programmer so I might see if I can create a PR to support 1.21 unless the author disapproves.

**Comment by apavarino** [comment]
Hello, thanks for the report, it's a good help for me to provide a stable 1.21 version. I'm currently working on it. Stay tunned !

**Comment by apavarino** [comment]
Seems to be a paper issue. I made few tests, deadchest handle well item meta and NBT tags. But something is different from  deachest block and game block. I compare values of ItemStack, this is strictly the same. i think it's something internal to the API. I will do more test when 1.21 Paper version will be more stable.
